### PR TITLE
Target net standard

### PR DESCRIPTION
**Pull request by sjp-canon**
Updates projects to new SDK-style.
Includes directory.build.props to allow for latest C# coding, .NET Standard targeting, and easy NuGet package management.
Modifies source files to adhere to standard coding style conventions (spaces for tabs, indenting, using statements inside namespace declarations, keyword spacing, etc).

### CODE REVIEW

**Comment by sudosilico** [comment]
Is there a reason you preferred `_ = sequence ?? throw new Exception();` over `if (sequence is null) throw new Exception();` aside from it being less characters?

**Comment by sjp-canon** [comment]
TL;DR - less characters and less discussion around style conventions.

After many years of schooling and professional work, the convention of braces around an if check (even a single line) was hammered into me for "better readability and coding style conventions". Remove the standard if check with syntactic sugar, remove the need to have the entire readability discussion. A small form of rebellion. Also, I like the null coalescing operator and try to use it so I get used to reading it in other's code.

Edit: if the use of the operator is bothersome, it's okay to replace with the original if check. @TChatzigiannakis has the ability, as do others with write access to the target branch, to alter the merge files.
